### PR TITLE
feat(protocol): check-in `HeklaTaikoToken`

### PR DESCRIPTION
**Pull request by davidtaikocha**
ref: https://holesky.etherscan.io/address/0x01bb2fd6d80942ce95b43c1322530fe690f2bc0e#code

### CODE REVIEW

File: packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20SnapshotUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "../../shared/common/EssentialContract.sol";
+
+/// @title HeklaTaikoToken
+/// @notice The HeklaTaikoToken (TTKOh), in the protocol is used for prover collateral
```
Comment (dantaik) [inline packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol:10]: how about we also use TKO as the symbol name?

Comment (dantaik) [reply packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol:10]: We can just add a function symbol() and return "TAIKO"

Comment (dantaik) [reply packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol:10]: Add this:

/// @dev Due to historical reasons, the Taiko Token on Hekla has a different storage layout compared to the mainnet token contract. Therefore, we need to maintain this file.

Comment (davidtaikocha) [reply packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol:10]: Added

**Review by YoGhurt111 — APPROVED** [review]

File: packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol:10]: > Add this:
> 
> /// @dev Due to historical reasons, the Taiko Token on Hekla has a different storage layout compared to the mainnet token contract. Therefore, we need to maintain this file.

Done

**Review by dantaik — APPROVED** [review]

**Review by YoGhurt111 — APPROVED** [review]

**Pull request merged**